### PR TITLE
Move away from string-based ID column as that causes problems on update

### LINKED ISSUE #1969: Mail not working after update to 0.16.0, exclamation marks left to account names
### Expected behavior
Two IMAP accounts were setup up with a previous version. They worked without a problem. 

### Actual behavior
All accounts won't work, instead of the mail folders only an exclamation mark is shown left to the account name. 
There is no output in the nextcloud log file

### Mail app

**Mail app version:** 0.16.0

**Mailserver or service:** Selft hosted, strato.de

### Server configuration

**Operating system**: Ubuntu 19.04 Host, official Docker image

**Web server:** Apache 2.4.38 as Proxy, official Docker image

**Database:** MariaDB 17.4 (docker)

**PHP version:** Official Docker Image

**Nextcloud Version:** 16.0.4, running in official Docker image

#### Client configuration
**Browser:** Firefox 68.0.2

**Operating system:** Ubuntu 19.04

### PR DESCRIPTION
**Pull request by ChristophWurst**
The QBMapper will automatically use int as column type for the `id` columns, hence an update on the old db schema failed with an sql error.

I've run into this before and yet I was stupid enough to make the same mistake again. Anyway, should be fine now. `name` is the new `id` and `id` is a simple auto increment column.

Fixes #1969

### CODE REVIEW

**Comment by ChristophWurst** [comment]
[mail.tar.gz](https://github.com/nextcloud/mail/files/3560289/mail.tar.gz)

**Comment by ChristophWurst** [comment]
This bug also prevented me from updating data for https://github.com/nextcloud/mail/pull/1970. Once this branch got merged into the other feature branch the error disappeared.

File: lib/Db/Mailbox.php [hunk 1]
```diff
@@ -29,6 +29,8 @@
 /**
  * @method string getId()
```
Comment (rullzer) [inline lib/Db/Mailbox.php]: This should also be updated

File: lib/Db/Mailbox.php [hunk 2]
```diff
@@ -29,6 +29,8 @@
...
  * @method void setId(string $id)
```
Comment (rullzer) [inline lib/Db/Mailbox.php]: and this

**Pull request merged**